### PR TITLE
Add ability to configure range fields & fix spatial audio on dropped WTs

### LINKED ISSUE #2: Make hearing distance configurable
As said in the title, it would cool have a config file where you can lower the hearing distance.

### PR DESCRIPTION
**Pull request by etrant**
resolve #2

### CODE REVIEW

**Comment by etrant** [comment]
Dropped WTs will now play from center, range needs to be configured since there's this weird peaking issue. But that is believed to be all configuration, which is customizable.

**Comment by Fummowo** [comment]
> Dropped WTs will now play from center, range needs to be configured since there's this weird peaking issue. But that is believed to be all configuration, which is customizable.

Wouldn't it make sense for all non local player walkies to come from the center? Or from the direction its in :)

**Comment by n33kos** [comment]
@wheatwarrior Thanks very much for helping with development on this mod!

Unfortunately I wasn't expecting activity on this repository and I pushed [version 1.4.3](https://github.com/n33kos/AlwaysHearActiveWalkie/commit/b44d9dd67954bb9cf96dd2ba0e84393e7774a6a0?diff=split&w=0) straight to main after you prepared this PR which caused some conflicts. I don't have a lot of time to dedicate to this mod. If you are able to rebase this PR onto `main` that would be a huge help.

Due to the complexity of testing this mod I want to be very careful with merging functional changes. I have a tough time with testing as I can't always get enough people together to verify edge cases with multiple users. I love a lot of the the changes you've made but I'm going to leave some comments with requests to potentially break it up in to more discrete PRs which will be a bit easier for me to test in isolation.

Also be aware Chizaruu is working on a readability refactor in this PR which may cause some more conflicts as its touching the same files. https://github.com/n33kos/AlwaysHearActiveWalkie/pull/5

File: Plugin.cs [hunk 1]
```diff
@@ -1,39 +1,64 @@
...
-    public class LCAlwaysHearWalkieMod : BaseUnityPlugin
+    public const string modGUID = "suskitech.LCAlwaysHearActiveWalkie";
+    public const string modName = "LC Always Hear Active Walkies";
+    public const string modVersion = "1.4.2";
+}
+
+[BepInPlugin(PluginInfo.modGUID, PluginInfo.modName, PluginInfo.modVersion)]
+public class Plugin : BaseUnityPlugin
+{
+    private ConfigEntry<float> configAudibleDistance;
```
Comment (n33kos) [inline Plugin.cs:19]: I'm not familiar with configs in bepinex. I'm assuming [this](https://docs.bepinex.dev/articles/user_guide/configuration.html) is the correct documentation for it? I'll need to make sure to update the readme with details on how to configure the mod.

File: Plugin.cs [hunk 2]
```diff
@@ -1,39 +1,64 @@
...
+
+        Log.LogInfo("\\ /");
+        Log.LogInfo("/|\\");
+        Log.LogInfo(" |----|");
+        Log.LogInfo(" |[__]| Always Hear Active Walkies");
+        Log.LogInfo($" |.  .| Version {PluginInfo.modVersion} Loaded");
+        Log.LogInfo(" |____|");
     }
-}
+}
```
Comment (n33kos) [inline Plugin.cs:64]: I'm loving all of these configuration changes and thanks for catching my namespace mistake. Would it be possible to break _only_ the configuration piece into a separate PR so we can test and integrate configurable distances separate from the spatial audio changes?

File: patches/StartOfRoundPatch.cs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            else
+            {
+                playerControllerB2.voicePlayerState.Volume = 1f;
+            }
+        }
+
+        return false; // Skip the original method
+    }
+}
+
```
Comment (n33kos) [inline patches/StartOfRoundPatch.cs:118]: As mentioned above I think it would be easiest to review and test if we broke the configuration changes into a separate PR from these spatial audio adjustments. I have a few thoughts/questions here as well:

1. I've read through this but haven't grokked what has changed (don't have the time to pull up the decompiler and compare at the moment). It looks to me very similar to the original function so I assume you made a relatively small change and have copied the majority of the original function over. Can you give me a description of what is changed?
2. I'm hoping with this mod to avoid overriding vanilla functions as much as possible. This should prevent breakage and regression if the original dev makes subtle logic updates to those functions. I'm curious to know what the issue was and how this corrects it, we may be able to implement the equivalent in a less destructive way.

File: Plugin.cs (hunk 2 above)
Comment (n33kos) [reply Plugin.cs:64]: I ended up with some time tonight so resolved the conflicts and integrated these config changes for version 1.4.4. I've attributed you in the README, thanks very much!
